src/SpikeCore/SpikeCore/Modules/UserManagementModule.cs

```diff
@@ -0,0 +1,174 @@
+
+namespace SpikeCore.Modules
+{
+    public class UserManagementModule : ModuleBase
+    {
+        private const string IrcLoginProvider = "IrcHost";
+
+        private static readonly List<string> RegularRoles = new List<string> {"regular"};
+        private static readonly List<string> OpRoles = new List<string> {"regular", "op"};
+        private static readonly Regex CommandRegex = new Regex(@"~\S+\s(list|show|add|remove)\s?(.*)?");
+
+        public override string Name => "Users";
+        public override string Description => "Provides user management features. Is only available to admins.";
+
+        public override string Instructions =>
+            "list | show <email> | add <email> <irc host> <prefix match?> <role> | remove <email>";
+
+        private readonly UserManager<SpikeCoreUser> _userManager;
+        private readonly SpikeCoreDbContext _context;
+
+        public UserManagementModule(UserManager<SpikeCoreUser> userManager, SpikeCoreDbContext context)
+        {
+            _userManager = userManager;
+            _context = context;
+        }
+
+        protected override bool AccessAllowed(SpikeCoreUser user)
+        {
+            return base.AccessAllowed(user) && user.isAdmin();
+        }
+
+        protected override async Task HandleMessageAsyncInternal(IrcPrivMessage request,
+            CancellationToken cancellationToken)
+        {
+            var commandMatch = CommandRegex.Match(request.Text);
+
+            if (commandMatch.Success)
+            {
+                var command = commandMatch.Groups[1].Value;
+                var details = commandMatch.Groups[2].Value;
+                var splitDetails = details.Length > 0 ? details.Split(" ") : new string[0];
+
+                if (command.Equals("list", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    await ListUsers(request);
+                }
+
+                // Single argument sub-methods.
+                if (splitDetails.Length == 1)
+                {
+                    if (command.Equals("show", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        await ShowUser(request, splitDetails[0]);
+                    }
+
+                    if (command.Equals("remove", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        await RemoveUser(request, splitDetails[0]);
+                    }
+                }
+
+                if (command.Equals("add", StringComparison.InvariantCultureIgnoreCase) && splitDetails.Length > 3)
+                {
+                    await CreateUser(request, splitDetails[0], splitDetails[1], splitDetails[2],
+                        splitDetails[3], cancellationToken);
+                }
+            }
+        }
+
+        private async Task ListUsers(IrcPrivMessage request)
+        {
+            var users = string.Join(", ",
+                _userManager.Users.Select(user => $"{user.Email}"));
```

Comment:
```suggestion
                _userManager.Users.Select(user => user.Email));
```

Reply:
Ah right. I originally had this formatted to include the UserName... before I realized that the UserName and Email needed to be the same haha. Good catch, thanks!